sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -248,7 +249,7 @@ private[hive] object HadoopTableReader extends HiveInspectors {
    * instantiate a HadoopRDD.
    */
   def initializeLocalJobConfFunc(path: String, tableDesc: TableDesc)(jobConf: JobConf) {
-    FileInputFormat.setInputPaths(jobConf, path)
+    jobConf.set("mapred.input.dir", StringUtils.escapeString(path.toString()))
```

Comment:
o, I see. `getPathStrings` does not really care if a comma is escaped or not... Can we use `public static void setInputPaths(Job job,  Path... inputPaths)`? I think it is better to avoid using `set` directly with a string key (using a method seems more robust).